src/garage/torch/algos/td3.py

```diff
@@ -0,0 +1,429 @@
+        policy_noise=0.2,
+        policy_noise_clip=0.5,
+        clip_return=np.inf,
+        policy_lr=_Default(1e-4),
+        qf_lr=_Default(1e-3),
+        policy_optimizer=torch.optim.Adam,
+        qf_optimizer=torch.optim.Adam,
+        num_evaluation_episodes=10,
+        steps_per_epoch=20,
+        start_steps=10000,
+        update_after=1000,
+    ):
+
+        self._env_spec = env_spec
+        action_bound = self._env_spec.action_space.high[0]
+        self._max_action = action_bound if max_action is None else max_action
+        self._action_dim = self._env_spec.action_space.shape[0]
+        self._tau = target_update_tau
+        self._discount = discount
+        self._reward_scaling = reward_scaling
+        self._exploration_noise = exploration_noise
+        self._policy_noise = policy_noise
+        self._policy_noise_clip = policy_noise_clip
+        self._clip_return = clip_return
+        self._replay_buffer_size = replay_buffer_size
+        self._min_buffer_size = min_buffer_size
+        self._buffer_batch_size = buffer_batch_size
+        self._grad_steps_per_env_step = grad_steps_per_env_step
+        self._update_actor_interval = update_actor_interval
+        self._steps_per_epoch = steps_per_epoch
+        self._start_steps = start_steps
+        self._update_after = update_after
+        self._num_evaluation_episodes = num_evaluation_episodes
+        self.max_episode_length = env_spec.max_episode_length
+
+        self._episode_policy_losses = []
+        self._episode_qf_losses = []
+        self._epoch_ys = []
+        self._epoch_qs = []
+        self._eval_env = None
+        self.exploration_policy = exploration_policy
+        self._uniform_random_policy = uniform_random_policy
+        self.worker_cls = FragmentWorker
+        self.sampler_cls = LocalSampler
+
+        self._replay_buffer = replay_buffer
+        self.policy = policy
+        self._qf_1 = qf1
+        self._qf_2 = qf2
+        self._target_policy = copy.deepcopy(self.policy)
+        self._target_qf_1 = copy.deepcopy(self._qf_1)
+        self._target_qf_2 = copy.deepcopy(self._qf_2)
+
+        self._policy_optimizer = make_optimizer(policy_optimizer,
+                                                module=self.policy,
+                                                lr=policy_lr)
+        self._qf_optimizer_1 = make_optimizer(qf_optimizer,
+                                              module=self._qf_1,
+                                              lr=qf_lr)
+        self._qf_optimizer_2 = make_optimizer(qf_optimizer,
+                                              module=self._qf_2,
+                                              lr=qf_lr)
+        self._actor_loss = torch.zeros(1)
+
+    def __getstate__(self):
+        """Object.__getstate__.
+
+        Returns:
+            dict: the state to be pickled for the instance.
+
+        """
+        data = self.__dict__.copy()
+        del data['_replay_buffer']
```

Comment:
Please don't do this, since it makes the results of `resume` really difficult to interpret. Just set the `snapshot_gap` (and `snapshot_mode`) parameters to `wrap_experiment`.

Reply:
@krzentner Do you mean by removing the line that set state by making a copy of `self.__dict__` (because it can be slow?? so it's better to use `snapshot_gap` instead)?

Reply:
I actually don't think you need `__setstate__` and `__getstate__` at all. It's true that pickling the replay buffer is slow, but that's not a big problem if we snapshot only occasionally.

Reply:
Got it, I agreed and it's fixed.